packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatus.tsx

```diff
@@ -155,44 +154,40 @@ const TextCointainer = styled.div`
 
 interface CoinjoinStatusProps {
     session: CoinjoinSession;
-    accountKey: string;
+    account: Account;
 }
 
-export const CoinjoinStatus = ({ session, accountKey }: CoinjoinStatusProps) => {
+export const CoinjoinStatus = ({ session, account }: CoinjoinStatusProps) => {
+    const accountKey = account.key;
     const [isWheelHovered, setIsWheelHovered] = useState(false);
     const sessionProgress = useSelector(state =>
         selectSessionProgressByAccountKey(state, accountKey),
     );
-    const coinjoinAccount = useSelector(state => selectCoinjoinAccountByKey(state, accountKey));
 
-    const { isCoinjoinResumeBlocked, coinjoinResumeBlockedMessageId, featureMessageContent } =
-        useBlockedCoinjoinResume();
+    const dispatch = useDispatch();
 
-    const actions = useActions({
-        stopCoinjoinSession,
-        pauseCoinjoinSession,
-        restoreCoinjoinSession,
-    });
-
-    const { isLocked } = useDevice();
+    const { coinjoinSessionBlocker, coinjoinSessionBlockedMessage } =
+        useCoinjoinSessionBlockers(account);
 
     const sessionPhase = useCoinjoinSessionPhase(accountKey);
     const menuRef = useRef<HTMLUListElement & { close: () => void }>(null);
     const theme = useTheme();
 
     const { paused, roundPhase, roundPhaseDeadline, sessionDeadline } = session;
     const isPaused = !!paused;
-    const isLoading = coinjoinAccount?.session?.starting;
-    const isToggleDisabled = isLoading || isLocked();
+    const isResumeBlocked = !!coinjoinSessionBlocker;
+    const isLoading = coinjoinSessionBlocker === 'sessionStarting';
+    const isResumeBlockedByExternalIssue =
+        isResumeBlocked && !['deviceLocked', 'sessionStarting'].includes(coinjoinSessionBlocker);
```

Comment:
Why red only because of these two?

Reply:
These two states are ephemeral and will happen quite frequently, so this prevents the colour changing unnecessarily. I renamed the variable to `isResumeBlockedByLastingIssue`.